### PR TITLE
Add support for 'None' on response data

### PR DESCRIPTION
**Pull request by avibhstarburst**
## Description
Add support for 'None' on response data 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: trino/client.py [hunk 1]
```diff
@@ -633,6 +633,9 @@ def process(self, http_response) -> TrinoStatus:
 
         self._next_uri = response.get("nextUri")
 
+        data = response.get("data", [])
```
Comment (hashhar) [inline trino/client.py:636]: Probably would be more readable/less surprising if you remove the 2nd empty list arg to response.get.

I understand what happens is that the "data" key exists with null value so instead of returning empty list it returns None.

**Review by hashhar — COMMENTED** [review]
@hovaesco can you guide with how to add a unit test for this (using mocking since not easy to have isolated test case).

File: trino/client.py (hunk 1 above)
Comment (hashhar) [reply trino/client.py:636]: Or maybe:

```
data = if response.get("data") else []
```

**Comment by hovaesco** [comment]
To write a unit test here please create mock like https://github.com/trinodb/trino-python-client/blob/ca38d2908533d891eedf5ee80f38dbdcfbbdb564/tests/unit/conftest.py#L64 and then see how it's used in tests https://github.com/trinodb/trino-python-client/blob/ca38d2908533d891eedf5ee80f38dbdcfbbdb564/tests/unit/test_client.py#L682

File: tests/unit/conftest.py [hunk 2]
```diff
@@ -184,6 +184,126 @@ def sample_get_response_data():
...
+                "name": "state",
+                "type": "varchar",
+                "typeSignature": {
+                    "typeArguments": [],
+                    "arguments": [{"kind": "LONG_LITERAL", "value": 2147483647}],
+                    "literalArguments": [],
+                    "rawType": "varchar",
+                },
+            },
+        ],
```
Comment (hashhar) [inline tests/unit/conftest.py:253]: shorten this to a single column for example, our test doesn't depend on so many being present.

**Review by hashhar — APPROVED** [review]
LGTM % comment.

**Review by hashhar — APPROVED** [review]
Thanks. Will merge after CI is done.
